spacy/cli/debug_data.py

```diff
@@ -247,6 +249,52 @@ def debug_data(
                     msg.warn(f"No examples for texts WITHOUT new label '{label}'")
                     has_no_neg_warning = True
 
+            with msg.loading("Obtaining span characteristics..."):
+                span_length = {
+                    label: _gmean(l)
+                    for label, l in gold_train_data["spans_length"][spans_key].items()
+                }
+                # Get relevant distributions: corpus, spans, span boundaries
+                p_corpus = _get_distribution(
+                    [eg.reference for eg in train_dataset], normalize=True
+                )
+                p_spans = {
+                    label: _get_distribution(spans, normalize=True)
+                    for label, spans in gold_train_data["spans_per_type"][
+                        spans_key
+                    ].items()
+                }
+                p_bounds = {
+                    label: _get_distribution(sb["start"] + sb["end"], normalize=True)
+                    for label, sb in gold_train_data["sb_per_type"][spans_key].items()
+                }
+                # Compute for the span characteristics
+                span_distinctiveness = {
+                    label: _get_kl_divergence(freq_dist, p_corpus)
+                    for label, freq_dist in p_spans.items()
+                }
+                sb_distinctiveness = {
+                    label: _get_kl_divergence(freq_dist, p_corpus)
+                    for label, freq_dist in p_bounds.items()
+                }
+
+            msg.info(f"Span characteristics for spans_key `{spans_key}`")
+            headers = (
+                "Span Type",
+                "Span Length",
+                "Span Distinctiveness",
+                "Boundary Distinctiveness",
+            )
+            metrics = [span_length, span_distinctiveness, sb_distinctiveness]
+            span_char = _compile_span_characteristics(
+                span_data=metrics,
+                labels=data_labels,
+            )
+            # Compute for weighted average (by frequency)
+            wgt_avg = [_wgt_average(metric, data_labels) for metric in metrics]
+            _wgt_avg_row = ["Wgt. Average"] + [str(w) for w in wgt_avg]
+            msg.table(span_char, footer=_wgt_avg_row, header=headers, divider=True)
```

Comment:
In the final table, I would round all numbers to two decimals, I don't think more is meaningful?

Reply:
Agree will do that!